contrib/python/podman/libs/containers.py

```diff
@@ -93,6 +94,41 @@ def export(self, target):
             results = podman.ExportContainer(self.id, target)
         return results['tarfile']
 
+    def commit(self, image_name, *args, **kwargs):
+        """Create image from container.
+
+        All changes overwrite existing values.
+          See inspect() to obtain current settings.
+
+        Changes:
+            CMD=/usr/bin/zsh
+            ENTRYPOINT=/bin/sh date
+            ENV=TEST=test_containers.TestContainers.test_commit
+            EXPOSE=8888/tcp
+            LABEL=unittest=test_commit
+            USER=bozo:circus
+            VOLUME=/data
+            WORKDIR=/data/application
+        """
+        try:
+            author = kwargs.get('author', getpass.getuser())
+        except Exception:
+            author = ''
+
+        changes = kwargs.get('changes', [])
+        for c in changes:
+            if c.startswith('LABEL') and c.count('=') < 2:
+                raise ValueError(
+                    'LABEL should have the format: LABEL=label=value, not {}'.
+                    format(c))
+        message = kwargs.get('message', '')
+        pause = kwargs.get('pause', True)
```

Comment:
For these two with fixed defaults, I'd rather drop the lines here and just make them arguments with defaults:

```python
def commit(self, image_name, message='', pause=True, **kwargs):
```

I've dropped `*args`, since you don't consume it anywhere.  If you decide you want to reinstate it, you can since [PEP 3102][1]:

```python
def commit(self, image_name, *args, message='', pause=True, **kwargs):
```

[1]: https://www.python.org/dev/peps/pep-3102/#specification

Reply:
I'm never sure about long function signatures. 

I went with second call and added changes as well since it has a known default. Left todo to review once all the Commit() bugs are fixed.  The upstream code here more in flux than previous work.